tensorflow/python/ops/nn_ops.py

```diff
@@ -1679,22 +1680,25 @@ def _swap_axis(logits, dim_index, last_index, name=None):
 
   # If dim is not the last dimension, we have to do a transpose so that we can
```

Comment:
Thanks for the PR! But can't we do something even more direct here?

This code already assumes that shape.ndims is not None, and that dim is a Python integer. So I don't think we need to do this test as part of the graph — we can do it in Python instead, which is simpler and easier to understand. i.e., I think you can just write:

```python
if dim < -shape.ndims or dim >= shape.ndims:
  raise ... etc.
```

This would also give the caller a more usable backtrace from Python, rather than a somewhat more cryptic error when running the graph.